sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2458,6 +2516,192 @@ checkDevSupportDeviceRequirements(const device &Dev,
     }
   }
 
+  // TODO: remove checks for CUDA and HIP from if-statement below when runtime
+  // query for them in matrix_combinations is implemented
+  if (JointMatrixPropIt &&
+      (Dev.get_backend() != sycl::backend::ext_oneapi_cuda) &&
+      (Dev.get_backend() != sycl::backend::ext_oneapi_hip)) {
+    std::vector<ext::oneapi::experimental::matrix::combination> Combinations =
+        Dev.get_info<
+            ext::oneapi::experimental::info::device::matrix_combinations>();
+
+    if (Combinations.empty())
+      return sycl::exception(errc::kernel_not_supported,
+                             "no matrix hardware on the target device, "
+                             "joint_matrix is not supported");
+
+    ByteArray JointMatrixByteArray =
+        DeviceBinaryProperty(*(JointMatrixPropIt.value())).asByteArray();
+    // Drop 8 bytes describing the size of the byte array.
+    JointMatrixByteArray.dropBytes(8);
+    std::string JointMatrixByteArrayToStr;
+    while (!JointMatrixByteArray.empty()) {
+      JointMatrixByteArrayToStr += JointMatrixByteArray.consume<char>();
+    }
+    std::istringstream JointMatrixStrStream(JointMatrixByteArrayToStr);
+    std::string SingleJointMatrix;
+
+    // start to parse the value which is generated by
+    // SYCLPropagateJointMatrixUsage pass
+    while (std::getline(JointMatrixStrStream, SingleJointMatrix, ';')) {
+      std::istringstream SingleJointMatrixStrStream(SingleJointMatrix);
+      std::vector<std::string> JointMatrixVec;
+      std::string Item;
+
+      while (std::getline(SingleJointMatrixStrStream, Item, ',')) {
+        JointMatrixVec.push_back(Item);
+      }
+
+      if (JointMatrixVec.size() == 4) {
+        std::string MatrixTypeUser = JointMatrixVec[0];
+        std::string UseStrUser = JointMatrixVec[1];
+        size_t RowsUser = std::stoi(JointMatrixVec[2]);
```

Comment:
Note: these two can throw exceptions if conversion can't be performed. I don't think that we should convert them to SYCL exceptions, but it could be useful if we want to have message saying "your device image is malformed", or if we want to silently ignore malformed device images, then we need to do a try-catch here.